File: app/widgets/taxa/image_gallery.js
//= require <templates/image_gallery>

JooseModule('Taxa', function () {
  JooseClass('ImageGallery', {
    isa: Templates.ImageGallery,
    has: {
      taxon          : { is: 'ro', required: true, nullable: false },
      joinClassname  : { is: 'ro', init: 'Taxon' },
      joinObj        : { is: 'ro', init: function () { return this.taxon() }},
      removeImagePath: { init: 'project_taxon_image_path' },
      galleryWidth   : { is: 'ro', init: 650 },
      galleryHeight  : { is: 'ro', init: 250 }
    },
    methods: {
      removeImagePath: function (id) {
        return this.route(this._removeImagePath, { taxon_id: this.params().id, id: id });
      }
    }
  })
});

File: app/widgets/taxa/image_gallery.rb
  module Taxa
    class ImageGallery < Templates::ImageGallery

      def initialize params
        params = {
          remove_image_path: 'project_taxon_image_path',
          join_classname:    'Taxon',
          join_obj:          params[:taxon] || fail('taxon required'),
          gallery_width:     650,
          gallery_height:    250
        }.merge(params)
        super
      end
    end
  end
